xrpl/utils/txn_parser/utils/balance_parser.py

```diff
@@ -0,0 +1,161 @@
+"""Helper functions for balance parser."""
+
+from decimal import Decimal
+from typing import Callable, List, Optional
+
+from pydash import group_by  # type: ignore
+
+from xrpl.utils.txn_parser.utils.nodes import NormalizedNode
+from xrpl.utils.txn_parser.utils.types import Balance, BalanceChange, BalanceChanges
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+
+def _get_xrp_quantity(
+    node: NormalizedNode,
+    value_parser: Callable[[NormalizedNode], Optional[Decimal]],
+) -> Optional[BalanceChange]:
```

Comment:
Is this method of this form (where it takes a method as a parameter) because you also need it for other transaction parser methods?

Reply:
Yes, `value_parser` can be the parser for balance changes or for final balances :)